### PR TITLE
Mk25

### PR DESCRIPTION
**Pull request by XPila**

### CODE REVIEW

File: Firmware/pins_Rambo_1_3.h [hunk 1]
```diff
@@ -18,85 +30,107 @@
...
+#define HEATER_2_PIN           -1
+#define TEMP_2_PIN             -1
 
-#define BEEPER                 84
+#define TEMP_AMBIENT_PIN        6 //A6
 
-#define BTN_EN1                72
-#define BTN_EN2                14
-#define BTN_ENC                 9
+#define TEMP_PINDA_PIN          3 //A3
```
Comment (3d-gussner) [inline Firmware/pins_Rambo_1_3.h:64]: As far i can see the '3 //A3' isn't connected to any connector should that be the free middle thermistor connector '1 //A1'

File: Firmware/pins_Rambo_1_3.h [hunk 2]
```diff
@@ -18,85 +30,107 @@
...
-  #define D_DATA               20
-  #define D_DATACLOCK          21
-#endif
+#define HEATER_0_PIN            3 //PE5
+#define TEMP_0_PIN              0 //A0
 
-// The SDSS pin uses a different pin mapping from file Sd2PinMap.h
-#define SDSS                   53
+#define HEATER_1_PIN           -1
+#define TEMP_1_PIN              1 //A1
```
Comment (3d-gussner) [inline Firmware/pins_Rambo_1_3.h:57]: Shouldn't that be -1 as I think this pin should be used for TEMP_PINDA_PIN?

File: Firmware/pins_Rambo_1_3.h [hunk 3]
```diff
@@ -18,85 +30,107 @@
...
-#define BEEPER                 84
+#define TEMP_AMBIENT_PIN        6 //A6
 
-#define BTN_EN1                72
-#define BTN_EN2                14
-#define BTN_ENC                 9
+#define TEMP_PINDA_PIN          3 //A3
 
-#define SDCARDDETECT           15
+#define VOLT_PWR_PIN            4 //A4
```
Comment (3d-gussner) [inline Firmware/pins_Rambo_1_3.h:66]: As far i can see the '4 //A4' isn't connected to any connector

File: Firmware/pins_Rambo_1_3.h [hunk 4]
```diff
@@ -18,85 +30,107 @@
...
+#define VOLT_BED_PIN            9 //A9
```
Comment (3d-gussner) [inline Firmware/pins_Rambo_1_3.h:67]: As far i can see the '9 //A9' isn't connected to any connector

File: Firmware/pins_Rambo_1_3.h [hunk 5]
```diff
@@ -18,85 +30,107 @@
...
 #define Y_MS1_PIN              69
 #define Y_MS2_PIN              39
+
 #define Z_STEP_PIN             35
 #define Z_DIR_PIN              47
 #define Z_MIN_PIN              10
 #define Z_MAX_PIN              23
 #define Z_ENABLE_PIN           27
 #define Z_MS1_PIN              68
 #define Z_MS2_PIN              67
```
Comment (3d-gussner) [inline Firmware/pins_Rambo_1_3.h:48]: Miss the MMU support
#ifdef SNMM
  #define E_MUX0_PIN           17
  #define E_MUX1_PIN           16
#endif

File: Firmware/pins_Rambo_1_3.h [hunk 6]
```diff
@@ -18,85 +30,107 @@
...
+#define LCD_PINS_D6         85
+#define LCD_PINS_D7         71
+
+//buttons are directly attached using AUX-2
+#define BTN_EN1                72
+#define BTN_EN2                14
+#define BTN_ENC                 9  // the click
+
+#define SDCARDDETECT           15
+
```
Comment (3d-gussner) [inline Firmware/pins_Rambo_1_3.h:109]: Missing TACH_0_PIN for the Noctua rpm sensor. As far i can see x-max, y-max and z-max are the last free pins can be used for that.
P1+P2 are used for LCD+Encoder+Beeper+SDcard
and P3 pins 3+5 are reserved for MMU
as well the P3 pins 6+8 for I2C (Filament Sensor)

File: Firmware/pins_Rambo_1_3.h [hunk 7]
```diff
@@ -18,85 +30,107 @@
...
+#define LOGIC_ANALYZER_CH2		53				// PB0 (PROC_nCS)
+// Channels 3-7 are slow, they could generate
+// 0.889Mhz waveform with a software loop and interrupt locking,
+// 1.333MHz waveform without interrupt locking.
+#define LOGIC_ANALYZER_CH3 		73				// PJ3
+// PK0 has no Arduino digital pin assigned, so we set it directly.
+#define WRITE_LOGIC_ANALYZER_CH4(value) if (value) PORTK |= (1 << 0); else PORTK &= ~(1 << 0) // PK0
+#define LOGIC_ANALYZER_CH5		16				// PH0 (RXD2)
+#define LOGIC_ANALYZER_CH6		17				// PH1 (TXD2)
+#define LOGIC_ANALYZER_CH7 		76				// PJ5
```
Comment (3d-gussner) [inline Firmware/pins_Rambo_1_3.h:127]: Not connected on the miniRAMBo

**Review by 3d-gussner — CHANGES_REQUESTED** [review]
Hope that helps a bit, even i would prefer a combined MK branch for MK1, MK2, MK2 MMU, MK2.5, MK2.5 MMU, MK3 and MK3 MMU

File: Firmware/pins_Rambo_1_3.h (hunk 5 above)
Comment (XPila) [reply Firmware/pins_Rambo_1_3.h:48]: hello, thanks for your comments. In future we will merge this branch back. Common branch for all printer variants will be better as you suggested.

File: Firmware/pins_Rambo_1_3.h (hunk 7 above)
Comment (XPila) [reply Firmware/pins_Rambo_1_3.h:127]: this pullrequest is just starting for MK2.5. Pinout is not final.

**Pull request merged**